hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -329,6 +316,23 @@ private void saveInternalSchema(HoodieTable table, String instantTime, HoodieCom
     }
   }
 
+  void runTableServicesInline(Option<Map<String, String>> extraMetadata, HoodieTable table, HoodieCommitMetadata metadata) {
+    // We don't want to fail the commit if hoodie.fail.writes.on.inline.table.service.exception is false. We catch warn if false
+    try {
+      // trigger clean and archival.
+      // Each internal call should ensure to lock if required.
+      mayBeCleanAndArchive(table);
+      // do this outside of lock since compaction, clustering can be time taking and we don't need a lock for the entire execution period
+      runCompactionClusteringInline(table, metadata, extraMetadata);
+    } catch (Exception e) {
+      if (config.isFailOnInlineTableServiceExceptionEnabled()) {
```

Comment:
Can we also enable cleaning for some existing test that validates the behavior due to this config? If this config is not being used in any test, then we should add a test (not a blocker, can do in a followup if no such test exists).